File: backend_analisis_exergetico/schemas/sustancia.py
from fastapi_utils.api_model import APIModel


class SustanciaBase(APIModel):
    nombre: str
    exergia: float


class SustanciaCreate(SustanciaBase):
    pass


class SustanciaUpdate(SustanciaBase):
    pass


class Sustancia(SustanciaBase):
    id: int

File: frontend_analisis_exergetico/src/modules/exergia/models/sustancia.ts
import { BaseModel } from 'src/utils/types';

export interface SustanciaBase extends BaseModel {
  nombre: string;
  exergia: number;
}

export type SustanciaCreateBase = SustanciaBase;
export type SustanciaUpdateBase = SustanciaBase;

export interface Sustancia extends SustanciaBase {
  id: number;
}
